source/uwp/SharedRenderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -367,314 +371,219 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering
-        httpBaseProtocolFilter.AllowUI(false);
-
-        winrt::HttpClient httpClient{httpBaseProtocolFilter};
+        auto isAutoSize{properties.isAutoSize};
+        auto isSVG{properties.isImageSvg};
+        auto weakThis{get_weak()};
+        auto weakImageSource{winrt::make_weak(imageSource)};
+        auto weakFrameworkElement = winrt::make_weak(properties.uiElement);
 
-        // Create an ImageSource to hold the image data.  We use BitmapImage & SvgImageSource in order to allow
-        // the tracker to subscribe to the ImageLoaded/Failed events
-        auto image = CreateImageSource(imgProperties.isImageSvg);
-        this->m_imageLoadTracker->TrackImage(image);
-
-        if (!imgProperties.isImageSvg)
+        try
         {
-            image.as<winrt::BitmapImage>().CreateOptions(winrt::BitmapCreateOptions::None);
-        }
-
-        auto getStreamOperation = httpClient.GetInputStreamAsync(imageUrl);
-        getStreamOperation.Completed(
-            [this, weakThis = this->get_weak(), imgProperties, image](
-                auto const& operation, auto status) -> void
+            // if image loading is handled by Xaml, we set this to true;
+            auto imageFiresOpenEvent{true};
+            // we can not do any works until stream is resolved
+            auto stream = co_await ResolveToStreamAsync(uri, resolvers, isSVG);
+            if (auto strongImageSource = weakImageSource.get())
             {
-                if (status == winrt::AsyncStatus::Completed)
+                if (stream)
                 {
-                    if (auto strongThis = weakThis.get())
+                    imageFiresOpenEvent = false;
+
+                    // the big differences between SVG and non-SVG images are:
+                    // 1. SVG size needs to be parsed from its data source
+                    // 2. SVG images need to be rasterized to a specific size
+                    if (isSVG)
                     {
-                        auto imageStream = operation.GetResults();
-                        winrt::InMemoryRandomAccessStream randomAccessStream{};
-                        auto copyStreamOperation = winrt::RandomAccessStream::CopyAsync(imageStream, randomAccessStream);
+                        auto sizeParseOp{ParseSizeOfSVGImageFromStreamAsync(stream)};
+                        // as size is being parsed, we can switch to the UI thread to set the source
+                        co_await wil::resume_foreground(GetDispatcher(strongImageSource));
+                        auto svgImageSource = strongImageSource.as<winrt::SvgImageSource>();
 
-                        m_copyStreamOperations.push_back(copyStreamOperation);
+                        if (!isAutoSize)
+                        {
+                            auto size = co_await sizeParseOp;
+                            svgImageSource.RasterizePixelHeight(size.Height);
+                            svgImageSource.RasterizePixelWidth(size.Width);
```

Comment:
This works for person styling as well? I remember we used to have an aspect ratio issue

Reply:
Just saw the example screenshots so I assume yes 😄

Reply:
yes, it works for person styling as well. please see the provided image in the description.